src/partitioning/parmetis_partitioner.C

```diff
@@ -203,12 +203,12 @@ struct SyncLocalIDs
   map_type & id_map;
 
   void gather_data (const std::vector<dof_id_type> & ids,
-                    std::vector<datum> & local_ids)
+                    std::vector<datum> & local_ids) const
   {
     local_ids.resize(ids.size());
 
     for (std::size_t i=0, imax = ids.size(); i != imax; ++i)
-      local_ids[i] = id_map[ids[i]];
+      local_ids[i] = id_map.at(ids[i]);
```

Comment:
If you switch to `map::at()`, you might want to think about `try`/`catch` block to go with it... this code will now throw instead of silently setting a default value for `local_ids[i]` in case `ids[i]` is not found for some reason.

Reply:
It's intended to throw, actually!  Silently setting a default value (which actually happened to me once in a previous even-buggier iteration of this refactoring) is a horrible result which leads to tricky bugs triggered within ParMETIS code later.  I could add a try / catch-then-libmesh_error block if you like, to make any internal errors even easier to debug.  But this is a spot where any error would truly be *internal*, not triggerable by an error in user code.

Reply:
I like catch-then-libmesh error personally, the more local the error information, the better.